packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -311,6 +313,16 @@ export class FeatureDevController {
         }
     }
 
+    private disposeToken(session: Session | undefined) {
```

Comment:
Can we add a comment on this function why its needed? I assume its because we want to refresh the token on every request? Or in this approach we only dispose it when we cancel something -- since otherwise the token will still be valid?

Reply:
Sure, I'll provide a more clearer comment there. Essentially when user stops he still in the same session, so we need clean up the token and create a new one for user provide a new prompt.